geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/io/FileSizeEstimator.scala

```diff
@@ -0,0 +1,70 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.io
+
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+ * Estimates how many features to write to create a file of a target size
+ *
+ * @param target target file size, in bytes
+ * @param error acceptable percent error for file size, in bytes
+ * @param estimatedBytesPerFeature initial estimate for bytes per feature
+ */
+class FileSizeEstimator(target: Long, error: Float, estimatedBytesPerFeature: Float) extends LazyLogging {
+
+  require(error >= 0 && error < 1f, "Error must be a percentage between [0,1)")
+
+  private val threshold = math.round(target * error.toDouble)
+  private var estimate = estimatedBytesPerFeature.toDouble
+  private var updatedEstimate: Option[Float] = None
+
+  /**
+   * Estimate how many features to write to hit our target size
+   *
+   * @param written number of bytes written so far
+   * @return
+   */
+  def estimate(written: Long): Int = {
+    val e = math.round((target - written) / estimate)
+    if (e < 1) { 1 } else if (e.isValidInt) { e.intValue() } else { Int.MaxValue}
```

Comment:
Two small things:
1.  Space after `Int.MaxValue`....:)
2.  Does this return an Int so that it can be used in control flow?  (E.g., there's no reasonable way to write a Longs-worth of features?)

Reply:
```suggestion
    if (e < 1) { 1 } else if (e.isValidInt) { e.intValue() } else { Int.MaxValue }
```

Reply:
right, it's used as a loop control variable. most of the time, it will re-calculate after looping for that many results. So you can write a longs-worth of values, but you have to reset the loop counter a few times. that's not true for some file formats (like parquet), because you can't get an accurate file size until you actually close the file. so in those cases, you wouldn't be able to write more than Int.MaxValue features. I would guess most of the uses of this are in while loops, so we could probably make it a long instead if you think it's worthwhile. At an pretty conservative 10 bytes per feature, using an int would still give you about 20GB max file size.

Reply:
i went ahead and updated it to a long